corgi/collectors/pnc.py

```diff
@@ -2,14 +2,71 @@
 from collections.abc import Mapping
 from typing import Any
 
+from corgi.core.models import Component
+
 logger = logging.getLogger(__name__)
 
 
-def parse_pnc_sbom(data: Mapping[str, Any]) -> int:
-    """Parse an SBOM from PNC and create components
-    Temporarily returns an int for testing purposes, to be
-    completed in CORGI-488"""
-    if "components" not in data:
-        raise ValueError("SBOM is missing component data")
+class SbomerSbom:
+    """Collector to parse SBOMs from PNC/Sbomer. See CORGI-488.
+    Future development should generalize this where applicable
+    to other middleware."""
+
+    def __init__(self, data: Mapping[str, Any]):
+        # Product information
+        root_meta = {
+            param["name"]: param["value"] for param in data["metadata"]["component"]["properties"]
+        }
+        self.product = root_meta["errata-tool-product-name"]
+        self.product_version = root_meta["errata-tool-product-version"]
+        self.product_variant = root_meta["errata-tool-product-variant"]
+
+        if "components" not in data:
+            raise ValueError("SBOM is missing component data")
+
+        self.components = {c["bom-ref"]: c for c in data["components"]}
+        # The root component is listed separately in metadata
+        self.components["root"] = data["metadata"]["component"]
+        for bomref, component in self.components.items():
+            self.components[bomref]["meta_attr"] = {}
+            if "redhat" not in component["purl"]:
+                # Community components won't have build info
+                self.components[bomref]["namespace"] = Component.Namespace.UPSTREAM
+            else:
+                self.components[bomref]["namespace"] = Component.Namespace.REDHAT
+                build_urls = {
+                    ref["comment"]: ref["url"]
+                    for ref in component["externalReferences"]
+                    if ref["type"] == "build-system"
+                }
+
+                if "pnc-build-id" in build_urls:
+                    self.components[bomref]["meta_attr"]["pnc_build_id"] = build_urls[
+                        "pnc-build-id"
+                    ].split("/")[-1]
+                if "brew-build-id" in build_urls:
+                    self.components[bomref]["meta_attr"]["brew_build_id"] = build_urls[
+                        "brew-build-id"
+                    ].split("/")[-1]
+
+            # Declared purl
+            self.components[bomref]["meta_attr"]["purl_declared"] = component["purl"]
+
+            # Related URL
+            website = [
+                ref["url"] for ref in component["externalReferences"] if ref["type"] == "website"
+            ]
+            if website:
+                self.components[bomref]["related_url"] = website[0]
```

Comment:
Yep, since component registry only supports one URL, this code only looks for one. There are other candidates that might make sense in the absence of a "website" type reference though, like VCS, issue trackers, mailing lists, or download locations. None of the provided purls have download_urls, but I should definitely try to find _something_ to use if website isn't available.

Reply:
Or, since I'm already storing the VCS url in meta where it's available, would it be better to save each of the types of external references in meta, and then also use one of those as `related_url` if there's no `website`?